pkg/apply/apply.go

```diff
@@ -74,9 +74,44 @@ func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstruct
 			}
 
 			return errors.Wrapf(err, "could not update object %s", objDesc)
-		} else {
-			log.Print("update was successful")
 		}
+		log.Print("update was successful")
+	}
+
+	return nil
+}
+
+// DeleteObject deletes an object in the apiserver
+func DeleteObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstructured) error {
+	name := obj.GetName()
+	namespace := obj.GetNamespace()
+	if name == "" {
+		return errors.Errorf("object %s has no name", obj.GroupVersionKind().String())
+	}
+	gvk := obj.GroupVersionKind()
+	// used for logging and errors
+	objDesc := fmt.Sprintf("(%s) %s/%s", gvk.String(), namespace, name)
+	log.Printf("reconciling %s", objDesc)
+
+	if err := IsObjectSupported(obj); err != nil {
```

Comment:
This function is for `Apply`, we should support removal for everything with Delete.